packages/medusa-file-minio/src/services/minio.js

```diff
@@ -1,36 +1,37 @@
-import fs from "fs"
+import stream from "stream"
 import aws from "aws-sdk"
-import { AbstractFileService } from '@medusajs/medusa'
+import fs from "fs"
+import { AbstractFileService } from "@medusajs/medusa"
+import { MedusaError } from "medusa-core-utils"
 
 class MinioService extends AbstractFileService {
-  
   constructor({}, options) {
     super()
 
     this.bucket_ = options.bucket
     this.accessKeyId_ = options.access_key_id
     this.secretAccessKey_ = options.secret_access_key
+    this.private_bucket_ = options.private_bucket
+    this.private_access_key_id_ =
+      options.private_access_key_id ?? this.accessKeyId_
+    this.private_secret_access_key_ =
+      options.private_secret_access_key ?? this.secretAccessKey_
     this.endpoint_ = options.endpoint
     this.s3ForcePathStyle_ = true
     this.signatureVersion_ = "v4"
   }
 
   upload(file) {
-    aws.config.setPromisesDependency(null)
-    aws.config.update({
-      accessKeyId: this.accessKeyId_,
-      secretAccessKey: this.secretAccessKey_,
-      endpoint: this.endpoint_,
-      s3ForcePathStyle: this.s3ForcePathStyle_,
-      signatureVersion: this.signatureVersion_,
-    }, true)
+    this.updateAwsConfig()
 
+    const parsedFilename = parse(file.originalname)
+    const fileKey = `${parsedFilename.name}-${Date.now()}${parsedFilename.ext}`
     const s3 = new aws.S3()
     const params = {
       ACL: "public-read",
       Bucket: this.bucket_,
```

Comment:
question: Shouldn't it be like other methods and checking to either use the private or the public bucket?

Reply:
relating to the comment above, the upload generally is targeted at the public bucket. We could include the possibility of private uploads with the upload method as well though. wdyt?

Reply:
I thing that it would be nice at least it gives the user the ability to choose since he can specify it in the configuration